xsl/fo/autoidx-kosek.xsl

```diff
@@ -124,8 +124,8 @@
         <xsl:apply-templates select="key('group-code', $key)[&scope;]
                                      [count(.|key('primary', &primary;)
                                      [&scope;][1])=1]"
-                             mode="index-primary">
-          <xsl:sort select="&primary;" lang="{$lang}"/>
+                             mode="index-primary-kosek">
+          <xsl:sort select="i:term-index(&primary;)" lang="{$lang}"/>
```

Comment:
I have't debugged code myself. But if my understanding is correct i:term-index() function would convert string of letters to string of space separated numbers where each number corresponds to code of letter from l:letter element in localization data. Using such value in xsl:sort will not sort strings in a right order.